src/pupil_labs/realtime_api/device.py

```diff
@@ -248,6 +252,14 @@ async def post_template_data(
             logger.debug(f"[{self}.get_template_data] Send data's template: {result}")
             return result
 
+    async def get_question_by_id(self, question_id: T.Union[str, UUID]):
+        self.template_definition = await self.get_template()
```

Comment:
Instead of making a new http call every time someone calls this function, I'd first check if the question id is in the current `self.template_definition` and only update it if that's not the case.

Reply:
```python
    async def get_question_by_id(self, question_id: T.Union[str, UUID]):
        def find_item(template_definition):
            return next(
                (
                    item
                    for item in template_definition.items
                    if str(item.id) == str(question_id)
                ),
                None,
            )

        item = find_item(self.template_definition) if self.template_definition else None

        if item:
            return item

        new_template_definition = await self.get_template()
        if self.template_definition is not new_template_definition:
            self.template_definition = new_template_definition
            return find_item(self.template_definition)

        return None
```

Something like this, where we check first if the question_id is in the template already set, if it is not set or not found (assuming you changed it in between), we fetch it and then try again